language-support/ts/daml-ledger/index.ts

```diff
@@ -953,6 +955,61 @@ class Ledger {
     return decode(partyInfoDecoder, json);
   }
 
+  /**
+   * Fetch a list of all package IDs from the ledger.
+   *
+   * @returns List of package IDs.
+   *
+   */
+  async listPackages(): Promise<PackageId[]> {
+    const json = await this.submit('v1/packages', undefined, 'get');
+    return decode(jtv.array(jtv.string()), json);
+  }
+
+  /**
+   * Fetch a binary package.
+   *
+   * @returns The content of the package as a [Blob], or a list of errors.
+   *
+   * [Blob]: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+   *
+   */
+  async getPackage(id: PackageId): Promise<{success: true; package: ArrayBuffer } | {success: false; errors: string[]}> {
+    const httpResponse = await fetch(this.httpBaseUrl + 'v1/packages/' + id, {
+      headers: { 'Authorization': 'Bearer ' + this.token },
```

Comment:
Perhaps worth factoring out the logic for Token -> Headers?

Reply:
Will do.